### PR TITLE
KLIB reader: better isolation of implementation details

### PR DESCRIPTION
**Pull request by ddolovov**

### CODE REVIEW

File: utilities/src/main/kotlin/org/jetbrains/kotlin/cli/utilities/InteropCompiler.kt [hunk 1]
```diff
@@ -72,12 +74,11 @@ fun invokeInterop(flavor: String, args: Array<String>): Array<String> {
     )
 
     val importArgs = allLibraries.flatMap {
-        val manifestProperties = it.manifestProperties
         // TODO: handle missing properties?
-        manifestProperties["package"]?.let {
-            val pkg = it as String
-            val headerIds = (manifestProperties["includedHeaders"] as String).split(' ')
-            val arg = "$pkg:${headerIds.joinToString(";")}"
+        val libraryReader = it
```
Comment (SvyatoslavScherbina) [inline utilities/src/main/kotlin/org/jetbrains/kotlin/cli/utilities/InteropCompiler.kt]: Maybe specify lambda parameter name explicitly?
```
val importArgs = allLibraries.flatMap { libraryReader ->
```

Comment (ddolovov) [reply utilities/src/main/kotlin/org/jetbrains/kotlin/cli/utilities/InteropCompiler.kt]: Will fix

File: extracted/konan.serializer/src/org/jetbrains/kotlin/konan/library/KonanLibrary.kt [hunk 2]
```diff
@@ -1,44 +1,64 @@
...
+    val bitcodePaths: List<String>
+    val includedPaths: List<String>
+
+    val targetList: List<String>
+
+    val dataFlowGraph: ByteArray?
+    val moduleHeaderData: ByteArray
+    fun packageMetadata(fqName: String): ByteArray
+
+    // FIXME: ddol: to be refactored into some global resolution context
```
Comment (olonho) [inline extracted/konan.serializer/src/org/jetbrains/kotlin/konan/library/KonanLibrary.kt]: Better use `FIXME(ddol): what` syntax.

Comment (ddolovov) [reply extracted/konan.serializer/src/org/jetbrains/kotlin/konan/library/KonanLibrary.kt]: OK. I will fix this with a separate commit.

**Pull request merged**